artistools/inputmodel/modelfromhydro.py

```diff
@@ -114,6 +114,20 @@ def get_snapshot_time_geomunits(pathtogriddata: Path) -> tuple[float, float]:
     return simulation_end_time_geomunits, mergertime_geomunits
 
 
+def scale_model_to_time(targetmodeltime_days, t_model_days, modeldata):
+    timefactor = targetmodeltime_days / t_model_days
+    modeldata.eval("pos_x_min = pos_x_min * @timefactor", inplace=True)
+    modeldata.eval("pos_y_min = pos_y_min * @timefactor", inplace=True)
+    modeldata.eval("pos_z_min = pos_z_min * @timefactor", inplace=True)
+    modeldata.eval("rho = rho * @timefactor ** -3", inplace=True)
+    print(
+        f"Adjusting t_model to {targetmodeltime_days} days (factor {timefactor}) "
+        "using homologous expansion of positions and densities"
+    )
+    t_model_days = targetmodeltime_days
+    return t_model_days, modeldata
```

Comment:
This function doesn't need to return anything. t_model_days is just the targetmodeltime_days argument back out again, and the modeldata DataFrame has been modified in place. Here is fine, or you could move it to inputmodel/inputmodel_misc.py

Reply:
It's either return t_model_days there or always update it the line after it's been called. Is it not better to return it?

Reply:
And it can't have been modified in place. It's called griddata below. Where I want to call it the df is called model and I don't want to call the df griddata in other parts of the code.